src/main/java/com/cryptomanager/services/InvestmentProjection.java

```diff
@@ -0,0 +1,21 @@
+package com.cryptomanager.services;
+
+public class InvestmentProjection {
+
+    public static double calculateInvestmentProjection(int coinAmount, double price, double growthRate, int months){
+        if(months <= 0) { return coinAmount*price; }
+        double totalValue = coinAmount*price;
+        //Calcula o valor total baseado na taxa de crescimento mensal e quantidade de meses investido
+        totalValue = totalValue*Math.pow(1+growthRate, months);
+        return totalValue;
+    }
+
+    public static int calculateInvestmentProjectionTime(int coinAmount, double price, double growthRate, double targetValue){
+        if(coinAmount <= 0 || price <= 0 || targetValue <= 0 || growthRate <= -1) { return 0; }
+        double totalValue = coinAmount*price;
+        if(targetValue < totalValue && growthRate > 0) { return 0; }
```

Comment:
o ideal não deveria ser (targetValue < totalValue || growthRate < 0)?
se a condição original for mantida pode ocorrer do crescimento ser negativo e o cálculo mesmo assim ser executado.